pkg/app/validate.go

```diff
@@ -0,0 +1,407 @@
+package app
```

Comment:
Apologies for not being clear enough. This is what I meant with the refactor:

* Fetching apps and app catalogs should still belong to the `data` package (e.g. `pkg/data/domain/app` for app-specific k8s functionality)
* Validating apps would be done here, in the `app` package, with the list of apps provided as input for the validation functions.

However, that might be a lot of work to do at once at this point, and maybe having to rework the logic, so we can maybe do it incrementally in other PRs.
But as for this PR, it believe it would be worth moving the k8s fetching logic to the `data` package, and providing the `app` and `appcatalog` data services as input for these validation functions. That would make it easier to rework later.

Please let me know what you think

Reply:
OK let me give it a try. I thought to skip doing that for now since the actual fetching logic is not as complex as in the cluster creation case (theres no provider specific things to be aware of in this case) 

but since it makes sense from a code organisation point of view, I'll give it a go